yb-voyager/cmd/exportData.go

```diff
@@ -817,19 +818,20 @@ func applyTableListFlagsOnFullListAndAddLeafPartitions(fullTableList []sqlname.N
 		//Apply include table list filter if present
 		includeTableList, err = applyFilterAndAddLeafTable(tableListViaFlag, "include")
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	} else {
 		//this is only for removing  the mid level partitioned table from fullTableList
 		//by passing false in `addAllLeafPartitions` boolean flag which  means this function won't the leaf parittions again it will just filter the list based on type
 		//i.e. only add if its a normal, root, or leaf table.
 		_, includeTableList, err = addLeafPartitionsInTableList(includeTableList, false)
 		if err != nil {
-			return nil, fmt.Errorf("error keeping only leaf and root tables: %v", err)
+			return nil, nil, fmt.Errorf("error keeping only leaf and root tables: %v", err)
 		}
 	}
-	//return the final table list generated from the command flags  table-list (include list) / exclude-table-list in this run
-	return sqlname.SetDifferenceNameTuples(includeTableList, excludeTableList), nil
+	//return the include and exclude table list generated from the command flags  table-list (include list) / exclude-table-list in this run
+
+	return includeTableList, excludeTableList, nil
```

Comment:
what's the motivation for this change?

Reply:
So we have this common function for applying table-list flags and adding leaf partitions to it in the first run or subsequent run. Previously, we were using the `SetDifferenceNameTuples ` directly in this function to return a filtered list from include and exclude, but now, for subsequent runs, we are using a name registry list with a name tuple having all source-target names populated if found but only one of them in case not found.
The problem is now for subsequent runs we need to use `SetDifferenceNameTuplesWithKey` to get a filtered list for without worrying about the full NameTuple.
So made this change to accommodate that and the caller will call `SetDifferenceNameTuplesWithKey` / `SetDifferenceNameTuplesWithKey`respectively .